File: backend/cantina/src/main/java/com/cantina/cantina/domain/services/CardapioService.java
package com.cantina.cantina.domain.services;

import com.cantina.cantina.domain.models.Cardapio;
import com.cantina.cantina.domain.models.dtos.CardapioDTO;

public interface CardapioService {

    void createCardapio(Cardapio cardapio);
    CardapioDTO getCardapio(Long cardapioId);

}

File: frontend/cantina/src/app/domain/services/cardapio.service.ts
import { Injectable } from '@angular/core';
import { Observable, map } from 'rxjs';
import { CardapioRepository } from 'src/app/data/repositories/cardapio.repository';
import { Cardapio } from '../models/cardapio.model';

@Injectable({
    providedIn: 'root'
})
export class CardapioService {

    constructor(private cardapioRepository: CardapioRepository) { }

    getCardapio(cardapioId: number): Observable<Cardapio> {
        return this.cardapioRepository.getCardapio(cardapioId).pipe(
            map((data: Cardapio)=> {
                return data;
            })
        );
    };
}
